servers/src/main/java/tachyon/master/TachyonMaster.java

```diff
@@ -102,23 +102,24 @@ public TachyonMaster(TachyonConf tachyonConf) {
     TachyonConf.assertValidPort(port, mTachyonConf);
     TachyonConf.assertValidPort(webPort, mTachyonConf);
 
-    String hostnameExternal = mTachyonConf.get(Constants.MASTER_HOSTNAME, "localhost");
-    // The master listens to the MASTER_HOSTNAME_LISTENING configuration option.
-    // MASTER_HOSTNAME_LISTENING defaults to MASTER_HOSTNAME if unspecified. If set to
-    // MASTER_HOSTNAME_LISTENING_WILDCARD, server listens to all addresses.
-    String hostnameListening =
-        mTachyonConf.get(Constants.MASTER_HOSTNAME_LISTENING, hostnameExternal);
-
-    InetSocketAddress address = new InetSocketAddress(hostnameExternal, port);
-    InetSocketAddress addressListening;
-    if (hostnameListening.equals(Constants.MASTER_HOSTNAME_LISTENING_WILDCARD)) {
-      addressListening = new InetSocketAddress(port);
-    } else {
-      addressListening = new InetSocketAddress(hostnameListening, port);
-    }
+    // String hostnameExternal = mTachyonConf.get(Constants.MASTER_HOSTNAME, "localhost");
+    // // The master listens to the MASTER_HOSTNAME_LISTENING configuration option.
+    // // MASTER_HOSTNAME_LISTENING defaults to MASTER_HOSTNAME if unspecified. If set to
+    // // MASTER_HOSTNAME_LISTENING_WILDCARD, server listens to all addresses.
+    // String hostnameListening =
+    // mTachyonConf.get(Constants.MASTER_HOSTNAME_LISTENING, hostnameExternal);
+    //
+    // InetSocketAddress address = new InetSocketAddress(hostnameExternal, port);
+    // InetSocketAddress addressListening;
+    // if (hostnameListening.equals(Constants.MASTER_HOSTNAME_LISTENING_WILDCARD)) {
+    // addressListening = new InetSocketAddress(port);
+    // } else {
+    // addressListening = new InetSocketAddress(hostnameListening, port);
+    // }
+    InetSocketAddress masterBindAddress = NetworkAddressUtils.getMasterBindAddress(mTachyonConf);
```

Comment:
In new design, the wildcard binding could be set via static property "tachyon.master.bind.host=0.0.0.0" at tachyon-default.properties or environment variable "$TACHYON_MASTER_BIND_HOST". So we should be able to remove previous code safely.